### PR TITLE
[2.7] [HttpFoundation] HttpCache refresh stale responses containing an ETag

### PR DESCRIPTION
**Pull request by maennchen**
| Q | A |
| --- | --- |
| Branch? | 2.7 |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | #19390, #6746 |
| License | MIT |
| Doc PR |  |

This PR is the replacement of #19391, which points at the wrong branch.

### CODE REVIEW

File: src/Symfony/Component/HttpKernel/HttpCache/HttpCache.php [hunk 1]
```diff
@@ -600,6 +600,9 @@ protected function lock(Request $request, Response $entry)
      */
     protected function store(Request $request, Response $response)
     {
+        if (!$response->headers->has('Date')) {
```
Comment (maennchen) [inline src/Symfony/Component/HttpKernel/HttpCache/HttpCache.php]: I wanted to explicitly call `setDate` to clarify what the code is doing even though one line containing the following would have the same effect:

```php
$response->getDate();
```

File: src/Symfony/Component/HttpKernel/Tests/HttpCache/HttpCacheTest.php [hunk 2]
```diff
@@ -181,6 +181,31 @@ public function testRespondsWith304OnlyIfIfNoneMatchAndIfModifiedSinceBothMatch(
         $this->assertEquals(304, $this->response->getStatusCode());
     }
 
+    public function testIncrementsMaxAgeWhenNoDateIsSpecifiedEventWhenUsingETag()
```
Comment (nicolas-grekas) [inline src/Symfony/Component/HttpKernel/Tests/HttpCache/HttpCacheTest.php]: you should add the time-sensitive annotation:
```
/**
 * @group time-sensitive
 */
```

Comment (nicolas-grekas) [inline src/Symfony/Component/HttpKernel/Tests/HttpCache/HttpCacheTest.php]: ok, nevermind, the annotation is already on the test class itself

**Comment by nicolas-grekas** [comment]
:+1:

**Comment by jakzal** [comment]
:+1:

**Comment by nicolas-grekas** [comment]
Thank you @maennchen.

**Pull request merged**